### PR TITLE
switch from SA key to workload identity to connect to GCS fuse

### PR DESCRIPTION
**Pull request by gree-gorey**
Issue #17.

### CODE REVIEW

File: scripts/cluster/start.sh [hunk 1]
```diff
@@ -57,13 +57,15 @@ if [ -z "${GSA_EMAIL}" ]; then
   done
 fi
 
-if ! kubectl get secret/gcsfuse-secrets --namespace kube-system; then
-  KEY_PATH=$(mktemp -d)/key.json
-  gcloud iam service-accounts keys create "${KEY_PATH}" --iam-account "${GSA_EMAIL}"
-  kubectl create secret generic gcsfuse-secrets --from-file="${KEY_PATH}" --namespace kube-system
-  rm -rf $(dirname "${KEY_PATH}")
+KSA_NAME="${GSA_NAME}"
```
Comment (sroettger) [inline scripts/cluster/start.sh:60]: Could the KSA_NAME be some fixed value like "gcsfuse_sa" so that we don't have to replace it in the yaml?

Comment (gree-gorey) [reply scripts/cluster/start.sh]: Yes, that's certainly better. Fixed.

File: scripts/cluster/start.sh [hunk 2]
```diff
@@ -57,13 +57,15 @@ if [ -z "${GSA_EMAIL}" ]; then
...
 
-if ! kubectl get secret/gcsfuse-secrets --namespace kube-system; then
-  KEY_PATH=$(mktemp -d)/key.json
-  gcloud iam service-accounts keys create "${KEY_PATH}" --iam-account "${GSA_EMAIL}"
-  kubectl create secret generic gcsfuse-secrets --from-file="${KEY_PATH}" --namespace kube-system
-  rm -rf $(dirname "${KEY_PATH}")
+KSA_NAME="gcsfuse-sa"
+
+if ! kubectl get serviceaccount/${KSA_NAME} --namespace kube-system; then
+  kubectl create serviceaccount --namespace kube-system ${KSA_NAME}
```
Comment (sirdarckcat) [inline scripts/cluster/start.sh:63]: Is the if necessary? Can we use the apply format instead?

Comment (sirdarckcat) [reply scripts/cluster/start.sh:63]: See the way the configmap is created (it pipes a create into an apply)

Comment (gree-gorey) [reply scripts/cluster/start.sh]: Yes, you are right. Changed for apply.

**Comment by sirdarckcat** [comment]
Looks good. Are we ok to merge it from your side?

**Comment by gree-gorey** [comment]
> Looks good. Are we ok to merge it from your side?

Yes, we can merge.

**Pull request merged**